packages/p2p/src/components/buy-sell/buy-sell-modal.jsx

```diff
@@ -36,6 +36,34 @@ BuySellModalFooter.propTypes = {
     onSubmit: PropTypes.func.isRequired,
 };
 
+const generateModalTitle = (formik_ref, my_profile_store, table_type, selected_ad) => {
+    if (my_profile_store.should_show_add_payment_method_form) {
+        if (!isMobile()) {
+            return (
+                <>
+                    <Icon
+                        icon='IcArrowLeftBold'
+                        onClick={() => {
+                            if (formik_ref.current.dirty) {
```

Comment:
Did not check for `my_profile_store.selected_payment_method` since it was clarified from design team that when user creates a new PM in buy/sell, the payment method is automatically selected and the cancel PM modal should only be shown when user edits the form (checked with `dirty` from Formik state)